### PR TITLE
Restructure Java staging clients

### PR DESCRIPTION
**Pull request by shinfan**
The new structure will make the CI depend on the local proto and grpc clients instead of depending on Maven. 

The new structure looks like:
- root
  - generated
    - java
      - settings.gradle
      - gradle wrapper
      - build.gradle
      - _gapic-* clients_
      - _proto-* clients_
        - build.gradle
      - _grpc-* clients_
        - build.gradle

In the root pom, the dependency configuration looks like this: (Use error reporting as an example)
```xml
project(':generated:java:gapic-google-cloud-error-reporting-v1beta1') {
  dependencies {
    compile project(':generated:java:proto-google-cloud-error-reporting-v1beta1')
    testCompile project(':generated:java:grpc-google-cloud-error-reporting-v1beta1')
  }
} 
```

### CODE REVIEW

**Comment by lukesneeringer** [comment]
Can the gradle specific stuff live in `generated/java` and not in the shared area?

**Comment by shinfan** [comment]
@lukesneeringer  I have moved the java stuff under `generated/java` and created an [issue](https://github.com/googleapis/api-client-staging/issues/298) to move PHP stuff under `generated/php`. PTAL.

File: generated/java/gapic-google-cloud-error-reporting-v1beta1/src/main/java/com/google/cloud/errorreporting/spi/v1beta1/ErrorGroupServiceClient.java [hunk 1]
```diff
@@ -87,7 +87,7 @@
  * </pre>
  */
 @Generated("by GAPIC")
-@ExperimentalApi
+@BetaApi
```
Comment (garrettjonesgoogle) [inline generated/java/gapic-google-cloud-error-reporting-v1beta1/src/main/java/com/google/cloud/errorreporting/spi/v1beta1/ErrorGroupServiceClient.java]: Could you do a separate regeneration of the Java clients so that this noise isn't mixed in with your restructuring?

**Comment by lukesneeringer** [comment]
@shinfan Thanks. :-)

File: generated/java/gapic-google-cloud-error-reporting-v1beta1/src/main/java/com/google/cloud/errorreporting/spi/v1beta1/ErrorGroupServiceClient.java (hunk 1 above)
Comment (shinfan) [inline generated/java/gapic-google-cloud-error-reporting-v1beta1/src/main/java/com/google/cloud/errorreporting/spi/v1beta1/ErrorGroupServiceClient.java]: Sure, the new clients requires updates on the gRPC release. The old structure depends on Maven so I will batch release a new version of gRPC/proto packages and create a separate PR after.

**Comment by shinfan** [comment]
@garrettjonesgoogle PTAL

File: generated/java/proto-google-longrunning-v1/staging.gradle [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+repositories {
+  mavenCentral()
+  mavenLocal()
+}
+
+dependencies {
+  compile "com.google.protobuf:protobuf-java:3.2.0"
+  compile "com.google.api:api-common:1.0.0-rc1"
+  compile "com.google.api.grpc:proto-google-common-protos:0.1.9"
```
Comment (garrettjonesgoogle) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Why is this depending on 0.1.9 of common-protos?

File: generated/java/proto-google-longrunning-v1/staging.gradle [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  main {
+    java {
+      srcDir 'src/main/java'
+    }
+  }
+}
+
+artifacts {
+  archives javadocJar, sourcesJar
+}
```
Comment (garrettjonesgoogle) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: newline?

File: generated/java/proto-google-longrunning-v1/staging.gradle [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+targetCompatibility = 1.7
+
+repositories {
+  mavenCentral()
+  mavenLocal()
+}
+
+dependencies {
+  compile "com.google.protobuf:protobuf-java:3.2.0"
+  compile "com.google.api:api-common:1.0.0-rc1"
```
Comment (garrettjonesgoogle) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Can we get this updated to `1.1.0`? (Fine to do in a different PR)

File: generated/java/proto-google-longrunning-v1/staging.gradle [hunk 5]
```diff
@@ -0,0 +1,42 @@
+buildscript {
```
Comment (garrettjonesgoogle) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Since you only have the staging.gradle files so that you can declare project dependencies instead of maven dependencies, you should actually be able use a single build.gradle, since at release time, gradle will convert the project dependencies into maven dependencies.

File: generated/java/proto-google-longrunning-v1/staging.gradle (hunk 3 above)
Comment (shinfan) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Obsolete.

File: generated/java/proto-google-longrunning-v1/staging.gradle (hunk 2 above)
Comment (shinfan) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Done

File: generated/java/proto-google-longrunning-v1/staging.gradle (hunk 4 above)
Comment (shinfan) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Done

File: generated/java/proto-google-longrunning-v1/staging.gradle (hunk 5 above)
Comment (shinfan) [inline generated/java/proto-google-longrunning-v1/staging.gradle]: Done

**Comment by shinfan** [comment]
@garrettjonesgoogle PTAL

File: generated/java/proto-google-longrunning-v1/build.gradle [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+repositories {
+  mavenCentral()
+  mavenLocal()
+}
+
+dependencies {
+  compile "com.google.protobuf:protobuf-java:3.2.0"
+  compile "com.google.api:api-common:1.1.0"
+  compile "com.google.api.grpc:proto-google-common-protos:0.1.11"
```
Comment (garrettjonesgoogle) [inline generated/java/proto-google-longrunning-v1/build.gradle]: It looks like there are more places that could have the project dependency (iam too)

Comment (shinfan) [inline generated/java/proto-google-longrunning-v1/build.gradle]: Done. PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM after tests pass
